trie/flatdb_sub_trie_loader.go

```diff
@@ -460,8 +472,10 @@ func (fstl *FlatDbSubTrieLoader) finaliseRoot(cutoff int) error {
 		fstl.accData.FieldSet = 0
 	}
 	fstl.groups = fstl.groups[:0]
-	fstl.subTries.roots = append(fstl.subTries.roots, fstl.hb.root())
-	fstl.subTries.Hashes = append(fstl.subTries.Hashes, fstl.hb.rootHash())
+	if fstl.hb.hasRoot() {
```

Comment:
I have removed this check here to make sure the size of array `subTries` is always the same as the size of input `dbPrefixes`. Otherwise `HookSubTries` won't be able to match them up properly. Please remove